src/other/precompile.jl

```diff
@@ -0,0 +1,3219 @@
+        Base.precompile(Tuple{Core.kwftype(typeof(DataFrames.Type)),NamedTuple{(:a, :b, :c),Tuple{Array{String,1},UnitRange{Int},UnitRange{Int}}},Type{DataFrame}})
+        Base.precompile(Tuple{Core.kwftype(typeof(DataFrames.Type)),NamedTuple{(:g, :x),Tuple{Array{Int,1},Array{Union{Missing, DataFrame},1}}},Type{DataFrame}})
+        Base.precompile(Tuple{typeof(DataFrames._combine_tables_with_first!),DataFrame,Tuple{Array{Int,1},Array{Int,1}},Array{Int,1},Int,Int,Function,GroupedDataFrame{DataFrame},Nothing,Tuple{Symbol,Symbol},Val{true}})
+        Base.precompile(Tuple{DataFrames.Reduce{typeof(Base.add_sum),Nothing,Nothing},Array{Float64,1},GroupedDataFrame{DataFrame}})
+        Base.precompile(Tuple{typeof(getindex),DataFrame,InvertedIndex{Int},All{Tuple{}}})
+        Base.precompile(Tuple{Core.kwftype(typeof(DataFrames.Type)),NamedTuple{(:x, :y, :x_first),Tuple{Array{Int,1},Array{Int,1},Array{Int,1}}},Type{DataFrame}})
+        Base.precompile(Tuple{typeof(DataFrames.compute_indices),Array{Int,1},Int})
+        Base.precompile(Tuple{typeof(DataFrames._combine_multicol),NamedTuple{(:function, :x1_length),Tuple{Int,Int}},Function,GroupedDataFrame{DataFrame},Nothing})
+        Base.precompile(Tuple{typeof(getindex),DataFrame,Colon,Cols{Tuple{Symbol,Symbol,Int}}})
+        Base.precompile(Tuple{typeof(DataFrames.do_call),typeof(prod),Array{Int,1},Array{Int,1},Array{Int,1},GroupedDataFrame{DataFrame},Tuple{Array{DataFrame,1}},Int})
+        Base.precompile(Tuple{typeof(DataFrames._combine_process_pair_astable),Bool,GroupedDataFrame{DataFrame},Dict{Symbol,Tuple{Bool,Int}},Array{DataFrames.TransformationResult,1},Nothing,Array{Symbol,1},Bool,Array{Int,1},Union{Function, Type},Tuple{Array{Int,1}}})
+        Base.precompile(Tuple{typeof(DataFrames._combine_with_first),NamedTuple{(:x1,),Tuple{Int}},Function,GroupedDataFrame{DataFrame},Tuple{Array{Union{Missing, Rational{Int}},1}},Val{false},Array{Int,1}})
+        Base.precompile(Tuple{typeof(Base.Broadcast.broadcasted),Function,Base.Broadcast.Broadcasted{DataFrames.DataFrameStyle,Nothing,typeof(+),Tuple{DataFrame,DataFrame}},Array{Float64,2}})
+        Base.precompile(Tuple{Type{DataFrame},Array{UnitRange,1},Symbol})
+        Base.precompile(Tuple{typeof(DataFrames._sortperm),SubDataFrame{DataFrame,DataFrames.Index,Array{Int,1}},Base.Sort.MergeSortAlg,DataFrames.DFPerm{Base.Order.ForwardOrdering,Tuple{SubArray{Union{Missing, String},1,PooledArrays.PooledArray{Union{Missing, String},UInt8,1,Array{UInt8,1}},Tuple{Array{Int,1}},false},SubArray{Union{Missing, String},1,Array{Union{Missing, String},1},Tuple{Array{Int,1}},false},SubArray{Union{Missing, String},1,Array{Union{Missing, String},1},Tuple{Array{Int,1}},false}}}})
+        Base.precompile(Tuple{Core.kwftype(typeof(DataFrames.DataFrame!)),NamedTuple{(:x, :y, :copycols),Tuple{Array{Int,1},Array{Int,1},Bool}},typeof(DataFrame!)})
```

Comment:
here is an example of `DataFrame!`. It is not the end of the world to have these deprecations. We will simply have to remove them later. The general procedure should be to remove test/deprecated.jl from test suite for generation of precompile statements.

I think this is minor so we can leave it as is for now.

Reply:
Yeah, as you want. If you spot more like these then maybe it's worth dropping them.

Reply:
OK - I will try to mark them (I will do suggestions)

Reply:
I didn't expect you'd find so many of them. I should have looked more carefully. :-)